Lib/glyphsLib/builder.py

```diff
@@ -695,17 +695,47 @@ def add_groups_to_ufo(ufo, kerning_groups):
 
 def build_gdef(ufo):
     """Build a table GDEF statement for ligature carets."""
-    carets = {}
+    bases, ligatures, marks, carets = set(), set(), set(), {}
     for glyph in ufo:
         for anchor in glyph.anchors:
             name = anchor.get('name')
             if name and name.startswith('caret_') and 'x' in anchor:
                 carets.setdefault(glyph.name, []).append(str(anchor['x']))
-    if not carets:
+        glyphinfo = glyphsLib.glyphdata.get_glyph(glyph.name)
+        if glyphinfo:
+            category, subCategory = glyphinfo.category, glyphinfo.subCategory
+        else:
+            category, subCategory = None, None
+        # Glyphs.app assigns glyph classes like this:
+        # * Base: any glyph that has an attaching anchor and is neither
+        #   classified as Ligature nor Mark using the definitions below;
+        # * Ligature: if subCategory is "Ligature" and the glyph has at least
+        #   one attachment anchor;
+        # * Mark: if category is "Mark" and subCategory is either "Nonspacing"
+        #   or "Spacing Combining";
+        # * Compound: never assigned by Glyphs.app.
+        # https://github.com/googlei18n/glyphsLib/issues/85
+        if subCategory == 'Ligature' and glyph.anchors:
```

Comment:
so, now you're adding ligatures to GDEF if they contain at least one anchor of any kind (`if glyph.anchors`).

But previously, you were classifying them as ligatures if they had an anchor whose name started with "caret_".

in https://github.com/googlei18n/glyphsLib/issues/99#issuecomment-275087699, @schriftgestalt said:

> Glyphs puts the ligatures in the GDEF if the glyph has caret_X anchors.

While in https://github.com/googlei18n/glyphsLib/pull/100#issuecomment-275195121, he said:

> The Liga class is set if the subCategory "Ligature" and it has at least one attachment anchor.

I don't know what he means exactly by "attachment" anchor, but I think the two statements above are a bit contradictory...